redhat-developer/odo
# Added .odoignore ignore feature to the odo push command.

fixes #1036 

Also --ignore flag was also added to the odo push command.

GetAbsGlobExps() was added converting relative glob expressions into absolute ones. Function signature of IsGlobExpMatch() was changed to work without the directory name of the component.

Signed-off-by: mik-dass <mrinald7@gmail.com>

## How to test changes?
1. Trigger `odo push` and check if the files in matching in .gitignore/.odoignore are pushed into the pod or not
2. Trigger `odo push` with the `--ignore` flag and check if the files are pushed into the pod or not
